src/clients/HubPoolClient.ts

```diff
@@ -779,33 +797,33 @@ export class HubPoolClient extends BaseAbstractClient {
   getLatestExecutedRootBundleContainingL1Token(
     block: number,
     chain: number,
-    l1Token: string
+    l1Token: EvmAddress
   ): ExecutedRootBundle | undefined {
     // Search ExecutedRootBundles in descending block order to find the most recent event before the target block.
     return sortEventsDescending(this.executedRootBundles).find((executedLeaf: ExecutedRootBundle) => {
       return (
         executedLeaf.blockNumber <= block &&
         executedLeaf.chainId === chain &&
-        executedLeaf.l1Tokens.some((token) => token.toLowerCase() === l1Token.toLowerCase())
+        executedLeaf.l1Tokens.some((token) => token.eq(l1Token))
       );
     });
   }
 
-  getRunningBalanceBeforeBlockForChain(block: number, chain: number, l1Token: string): TokenRunningBalance {
+  getRunningBalanceBeforeBlockForChain(block: number, chain: number, l1Token: EvmAddress): TokenRunningBalance {
     const executedRootBundle = this.getLatestExecutedRootBundleContainingL1Token(block, chain, l1Token);
 
     return this.getRunningBalanceForToken(l1Token, executedRootBundle);
   }
 
   public getRunningBalanceForToken(
-    l1Token: string,
+    l1Token: EvmAddress,
     executedRootBundle: ExecutedRootBundle | undefined
   ): TokenRunningBalance {
     let runningBalance = toBN(0);
     if (executedRootBundle) {
       const indexOfL1Token = executedRootBundle.l1Tokens
-        .map((l1Token) => l1Token.toLowerCase())
-        .indexOf(l1Token.toLowerCase());
+        .map((_l1Token) => _l1Token.toAddress().toLowerCase())
+        .indexOf(l1Token.toAddress().toLowerCase());
```

Comment:
`Address.toAddress()` returns a checksummed address so it should be OK to drop the `.toLowerCase()` here.

Reply:
True. https://github.com/across-protocol/sdk/pull/904/commits/16dbd7e71e3f07706b0b708ea5e94aba544c84f5